README.md

```diff
@@ -54,6 +54,50 @@ pyyaml
 urllib3
 ```
 
+### Developer setup
+
+Clone the repository using `git`:
+```
+git clone https://github.com/zowe/zowe-client-python-sdk.git
+```
+
+We recommend that developers create a virtual environment to install all the required dependencies for this project.  
+**Note:** Use of `venv`, the virtual environment module, requires at least version 3.3 of Python.
+
+To create a virtual environment, make a folder to store your virtual environment (referred to below as `YOUR_VENV_PATH`). Then, use the `venv` module to set it up:
+
+```
+python3 -m venv YOUR_VENV_PATH
+```
+
+Next, activate your virtual environment so that Python uses this folder to manage dependencies.
+
+Assuming that you are using Bash shell, reference the command below. Otherwise, check the table titled "Command to activate virtual environment" [here](https://docs.python.org/3/library/venv.html#how-venvs-work) to find the command that works for your shell.
+
+```
+source YOUR_VENV_PATH/bin/activate
+```
+
+Navigate to the root of the repository and checkout the desired branch. Currently, active development is on the `main` branch.
```

Comment:
Should this step be moved above creating a virtual environment?

I think the `venv` is typically created in a subfolder of the repository (not required to do so, but probably recommended).

Reply:
For people who are new to Python and may not know what to supply for `YOUR_VENV_PATH`, should we suggest a default of either `venv` or `.venv` in the repo folder?

Reply:
I will update the instructions to create a `venv` folder in the repo folder, since its `gitignore`'d it should be fine 😋 

I think I'll go with `venv`